File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii-Solution1.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var deleteDuplicates = function(head) {
    const starter = new ListNode();
    let lastNode = starter;
    let currNode = head;

    while(currNode != null){
        let traverseNode = currNode.next;
        let currValCounter = 1;

        while(traverseNode != null && traverseNode.val == currNode.val){
            currValCounter++;
            traverseNode = traverseNode.next;
        }

        if(currValCounter == 1){
            lastNode.next = currNode;
            lastNode = currNode;
        }

        currNode = traverseNode;
    }

    lastNode.next = null;

    return starter.next;
};

File: 0082-remove-duplicates-from-sorted-list-ii/0082-remove-duplicates-from-sorted-list-ii-Solution1.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode DeleteDuplicates(ListNode head) {
        var starter = new ListNode();
        var lastNode = starter;
        var currNode = head;

        while(currNode != null){
            var traverseNode = currNode.next;
            var currValCounter = 1;

            while(traverseNode != null && traverseNode.val == currNode.val) {
                currValCounter++;
                traverseNode = traverseNode.next;
            }

            if(currValCounter == 1) {
                lastNode.next = currNode;
                lastNode = currNode;
            }
            
            currNode = traverseNode;
        }

        lastNode.next = null;

        return starter.next;
    }
}